tests/drivers/rtc/rtc_api/src/test_alarm.c

```diff
@@ -87,61 +109,79 @@ ZTEST(rtc_api, test_alarm)
 	/* Validate alarms supported fields */
 	for (uint16_t i = 0; i < alarms_count; i++) {
 		ret = rtc_alarm_get_supported_fields(rtc, i, &alarm_time_mask_supported);
-
 		zassert_ok(ret, "Failed to get supported alarm %d fields", i);
 
-		/* Skip test if alarm does not support the minute and hour fields */
-		if (((RTC_ALARM_TIME_MASK_MINUTE & alarm_time_mask_supported) == 0) ||
-		((RTC_TEST_ALARM_TIME_HOUR & alarm_time_mask_supported) == 0)) {
-			ztest_test_skip();
```

Comment:
Why get rid of ztest_test_skip()? Things now show up as failed if they are not supported which is not correct.
Also, the Ambiq now comes up with:
"Configured alarm time fields to set are not supported"

Reply:
The skip was only there because the alarm mask could not be configured :) Now that the mask is explicitly set for each RTC (or keeps the default mask), the test asserts that the alarm mask is valid.